docs/source/specs/typespec/main.tsp

```diff
@@ -0,0 +1,170 @@
+    }
+    
+    alias Problem500 = ProblemDetails<Status = 500>;
+
+    @error
+    model Problem400 extends ProblemDetails<Status = 400>{
+        detail: "JSON parse error - Expecting value: line 1 column 1 (char 0)",
+    }
+
+    alias Problem400AlreadyExists = ProblemDetails<Status = 400>;
+    
+    alias CommonProblems = Problem401 | Problem403 | Problem500;
+
+    namespace Workspace {
+        model Problem400WorkspaceNotEmpty extends Problems.ProblemDetails<Status = 400>{
+            title: "Unable to delete due to workspace dependencies",
+        }
+    }
+}
+
+@route("/workspaces")
+@tag("Workspaces")
+@doc("Operations about workspaces")
+namespace Workspaces {    
+    model BasicWorkspace {
+        @doc("Workspace A")
+        name: string = "Workspace A";
+        @doc("Description of Workspace A")
+        description?: string = "Description of Workspace A";
+    }
+
+    model Workspace {
+        @key uuid: UUID;
+        parent?: UUID;
+        ...BasicWorkspace;
+        ...Timestamps;
+    }
+
+    alias CreateWorkspaceRequest = BasicWorkspace;
+
+    model CreateWorkspaceResponse extends BasicWorkspace{
+        @statusCode _: 204;
+    }
+
+    model GetWorkspaceResponse extends BasicWorkspace{
+        @statusCode _: 200;
+    }
+
+    model PutWorkspaceRequest {
+        @doc("Workspace A")
+        name: string = "Workspace A";
+        @doc("Description of Workspace A")
+        description: string = "Description of Workspace A";
+    }
+
+    model PatchWorkspaceRequest {
+        @doc("Workspace A")
+        name?: string = "Workspace A";
+        @doc("Description of Workspace A")
+        description?: string = "Description of Workspace A";
+    }
+
+    model PutWorkspaceResponse extends BasicWorkspace{
+        @statusCode _: 201;
+    }
+
+    model PatchWorkspaceResponse extends BasicWorkspace{
+        @statusCode _: 204;
+    }
+
+    model DeleteWorkspaceResponse{
+        @statusCode
+        statusCode: 201,
+    }
+   
+    @doc("Create workspace in tenant")
+    @summary("Create workspace in tenant")
+    @post op create(@body body: CreateWorkspaceRequest): CreateWorkspaceResponse | Problems.CommonProblems | Problems.Problem400;
+
+    @doc("Get a workspace in tenant")
+    @summary("Get a workspace in tenant")
+    @get op read(
+        @doc("Unique identification")
+        @path uuid: UUID;
+    ): GetWorkspaceResponse | Problems.CommonProblems | Problems.Problem404;
+
+    @doc("Edit a workspace")
+    @summary("Edit the workspace name or description")
+    @put op update(
+        @doc("Unique identification")
+        @path uuid: UUID;
+        @body body: PutWorkspaceRequest
+    ): PutWorkspaceResponse | Problems.CommonProblems | Problems.Problem400AlreadyExists ;
+
+    @patch op patch(
+        @doc("Unique identification")
+        @path uuid: UUID;
+        @body body: PutWorkspaceRequest
+    ): PatchWorkspaceResponse | Problems.CommonProblems | Problems.Problem400AlreadyExists ;
```

Comment:
IMO I would just omit this API, there isn't a UI for it that I know of. Not a big deal tho.